### PR TITLE
[NTV-121] Post Comment Query Conversion To Apollo

### PR DESCRIPTION
**Pull request by msadoon**
# 📲 What and 🤔 Why

As part of our ongoing technical debt/migration from DSL/support of Apollo this is the conversion of the previous call `postComment` to Apollo away from our DSL mutation.

# 🛠 How

Used datadog to see old mutation logs for `postComment` service call and wrote `PostComment.graphql` mutation to replace it along with adapters.

# 🏎 Performance

- Identical to previous DSL request.

# ✅ Acceptance criteria

- [x] Add comment in comment feed is successful.
- [x] Add comment in reply feed is successful.

### CODE REVIEW

File: KsApi/models/graphql/adapters/Comment+PostCommentMutationData.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+extension Comment {
+  /**
+   Returns a minimal `Comment` from a `GraphAPI.PostCommentMutation.Data.CreateComment.Comment`
+   */
+
+  static func from(_ data: GraphAPI.PostCommentMutation.Data.CreateComment.Comment) -> Comment? {
+    guard let author = data.author,
+      let decomposedAuthorId = decompose(id: author.id),
+      let authorBadges = data.authorBadges else {
```
[the remark this answers is not in the document]
Comment (singhhari) [reply KsApi/models/graphql/adapters/Comment+PostCommentMutationData.swift:11]: We wouldn't want all three unwraps here to fail because of that.

**Review by singhhari — APPROVED** [review]
Any way to bump the coverage a bit before merging?

File: KsApi/models/graphql/adapters/Comment+PostCommentMutationData.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (msadoon) [reply KsApi/models/graphql/adapters/Comment+PostCommentMutationData.swift:11]: good point.

**Comment by msadoon** [comment]
> Any way to bump the coverage a bit before merging?

Sometimes I wonder if its' just misreporting these stats, because it says 86 hits, 19 misses and 3 partials. So there's clearly reporting more coverage than missed, not sure why this translates to lower overall coverage. In this case, lets' just let it go with -0.01%, to me that's basically 0.

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**